### PR TITLE
Scheduler: remove reflect.DeepEqual for defaultpreemption, helper, imagelocality package

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Use cmp instead of reflect.DeepEqual in scheduler test for defaultpreemption, helper and imagelocality package.

**Which issue(s) this PR fixes**:

Xref #94696

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/helper/spread_test.go [hunk 1]
```diff
@@ -97,8 +97,9 @@ func TestGetPodServices(t *testing.T) {
 			get, err := GetPodServices(fakeInformerFactory.Core().V1().Services().Lister(), test.pod)
 			if err != nil {
 				t.Errorf("Error from GetPodServices: %v", err)
-			} else if !reflect.DeepEqual(get, test.expect) {
-				t.Errorf("Expect services %v, but got %v", test.expect, get)
+			}
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/helper/spread_test.go:100]: leave the `else`. If there was an error, `get` would just be empty and not match.

File: pkg/scheduler/framework/plugins/imagelocality/image_locality_test.go [hunk 2]
```diff
@@ -348,8 +348,8 @@ func TestImageLocalityPriority(t *testing.T) {
 				gotList = append(gotList, framework.NodeScore{Name: nodeName, Score: score})
 			}
 
-			if !reflect.DeepEqual(test.expectedList, gotList) {
-				t.Errorf("expected:\n\t%+v,\ngot:\n\t%+v", test.expectedList, gotList)
+			if diff := cmp.Diff(test.expectedList, gotList); diff != "" {
+				t.Errorf("Unexpected node score list (-want, +got): %s", diff)
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/imagelocality/image_locality_test.go:352]: prefer leaving an empty line after `:`

File: pkg/scheduler/framework/plugins/helper/spread_test.go (hunk 1 above)
Comment (gavinfish) [reply pkg/scheduler/framework/plugins/helper/spread_test.go]: done.

File: pkg/scheduler/framework/plugins/imagelocality/image_locality_test.go (hunk 2 above)
Comment (gavinfish) [reply pkg/scheduler/framework/plugins/imagelocality/image_locality_test.go]: done.

**Pull request merged**